python/cugraph/components/cc_wrapper.pyx

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from c_cc cimport *
+#from libcpp cimport bool
+from c_graph cimport *
+#from libcpp cimport cugraph_cc_t
+from libc.stdint cimport uintptr_t
+import cudf
+#from pygdf import Column
+import numpy as np
+
+cpdef weak_cc(G, connect_type = CUGRAPH_WEAK):
```

Comment:
Can't find the matching Nx API function there https://networkx.github.io/documentation/stable/reference/algorithms/component.html

Reply:
Is it `connected_components(G)` or weakly_connected_components(G)?
It seems there's a typo in the latter as Nx doc says it works only for directed graphs.

Reply:
Nevermind, I found `weakly_connected_components(G)` below in the test.
The API should be upgraded to match Nx. I am still confused about the directed statement in Nx doc but I'm pretty sure they made a mistake.

Reply:
I am calling `nx.weakly_connected_components(Gnx)`. What exactly is the problem here?

Reply:
1. cugraph API is `weak_cc(G)` instead of `weakly_connected_components(G)` (or `connected_components(G)`)
2. nx.weakly_connected_components(Gnx) doc says it works only for directed graphs while yours works only for undirected graphs. This sounds like a mistake from Nx but you may want to double check. 
3. Is your feature closer to `nx.connected_components(G)` or `weakly_connected_components(G)`?

Reply:
The feature is closer to `weakly_connected_components(G)`. For the time being.

Reply:
1. update the name to be `weakly_connected_components `
2. update the doc to state what input is supported

Reply:
2. @afender, we discussed the API extensively. Stating in the doc that the input is supposed to be undirected, only nixes the logic of having the `connect_type` argument. The `connect_type` can be CUGRAPH_WEAK (default) or CUGRAPH_STRONG (currently not supported; for the future, when weak_cc will provide a version that works on directed graphs). I can add a comment that  CUGRAPH_STRONG is _currently_ not supported. But...requiring the graph to be undirected does not make sense with this API. And as said, _now_ is _not_ the time to change the C++ API. You had this opportunity when the API was discussed. And later, before the C++ PR got merged.

Reply:
Andrei, I am just talking about documentation and renaming here. This is not about building another API. 

I think it is important to tell the user what type of graph input is supported by your feature (edge list, directed/undirected, int32 indices). I'm sorry but I don't get why this can be a problem. I am actually upgrading the current doc of cugraph to better document what graph input are supported for each analytics as users got confused with that in the past. 

@BradReesWork please arbitrate.

Reply:
Weakly connected component is traditionally against an undirected graph - it is useful for finding disjointed components.  Running Weakly CC on a directed graph does require the code to add all the reverse edges.

I think we should match the Nx API of having the algorithm call:  `weakly_connected_components`

The output should be a DataFrame similar to what is returned with clustering algorithms for consistency:  ['vertex'] and ['component']

Once the `Graph` class is updated, it should include a flag for teh type of graph

Reply:
@afender, please provide the exact comment that you'd like to see here. I'm done guessing.

Reply:
This is the comment I added for Jaccard which I think would apply here : 
```
cuGraph graph descriptor, should contain the connectivity information as an edge list 
      (edge weights are not used for this algorithm). Currently, the graph should be undirected where 
      an undirected edge is represented by a directed edge in both directions.
      The adjacency list will be computed if not already present.
```

I have this under 
```
    Parameters
    ----------
    graph : cuGraph.Graph 
```

Feel free to improve this or specialize it for WCC

Reply:
Okay, fixed the comments.